tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -680,21 +709,67 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
       }
       return transp_dims;
     };
+    
+    // Plain transpose on a or b. Plain transposes a matrix by permuting its 
+    // dimension without changing storage order.
+    auto plain_transpose = [&](const char matrix_name) {
```

Comment:
Calling this with the operand and its dimensions instead of the name would make the if statement redundant.

Reply:
Fixed.